src/NBitcoin/NetworkOptions.cs

```diff
@@ -0,0 +1,149 @@
+﻿namespace NBitcoin
+{
+    /// <summary>
+    /// The current use-case for this class is to provide option-dependent serialization.
+    /// It is a drop-in replacement for the legacy TransactionOptions enumeration.
+    /// 
+    /// The NetworkOptions instance is intended to be relayed from the Network object to 
+    /// the child objects: Network => Block => Transaction => TxIn/TxOut => Coin => ...
+    /// It can also be used on repository or stream objects that need to deserialize the 
+    /// above objects or passed as a parameter futher down the call-hierarchy as needed.
+    /// </summary>
+    public class NetworkOptions
+    {
+        public const uint None = 0x00000000;
+        public const uint Witness = 0x40000000;
+        public const uint All = Witness;
+
+        //TODO?: Could be used by Block:
+        //public Action<NetworkOptions, Block> SetBlockSpecificFlags { get; set; } = null;
+        //TODO?: Could be used by Transaction:
+        //public Action<NetworkOptions, Transaction> SetTransactionSpecificFlags { get; set; } = null;
+
+        private uint flags = All;
```

Comment:
While its interesting to have a flag to represent all kind of options, we could easily solve this using a bool flag for witness right?

`  private bool Witness = false;`

Not saying thats the best way, also your approach has minimal changes.

But if later we have POS, we can just have a flag that say IsPos instead of another const
```
        public const uint None = 0x00000000;
        public const uint Witness = 0x40000000;
        public const uint Pos = 0x80000000;
        public const uint All = Witness + Pos ;
```

Wont this be easier in the long run?
```
        private bool Witness = false;
        private bool Pos = false;
```

Reply:
The uint does have some advantages in terms of mask based operarions - e.g. if we decide to handler serialization via field registrations then we could have masks on the flags that dictate when serialization should take place.

Also:

- the uint approach means we don't have to make any code changes
- we can support the boolean operations via get/set properties while still storing the underlying value in the flags variable.

Reply:
Yeah this is the good of both options.